cudamapper/src/chainer_utils.cu

```diff
@@ -0,0 +1,231 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "chainer_utils.cuh"
+
+#include <fstream>
+#include <sstream>
+#include <cstdlib>
+
+// Needed for accumulate - remove when ported to cuda
+#include <numeric>
+#include <limits>
+
+#include <cub/cub.cuh>
+#include <thrust/execution_policy.h>
+
+#include <claraparabricks/genomeworks/utils/cudautils.hpp>
+
+namespace claraparabricks
+{
+
+namespace genomeworks
+{
+
+namespace cudamapper
+{
+namespace chainerutils
+{
+
+__host__ __device__ Overlap create_simple_overlap(const Anchor& start, const Anchor& end, const int32_t num_anchors)
```

Comment:
I would expect that there is a very similar function somewhere in overlapper? Can it be reused?

Reply:
There is indeed, but I think it's reliant on the anchor chain being continuous:
https://github.com/clara-parabricks/GenomeWorks/blob/9a2e064b274ce7e6739fdc8a214fa7bd03f17ac7/cudamapper/src/overlapper_triggered.cu#L184-L233

I agree I should refactor both of these, though I think I need both because don't want to pass an array of anchors (and the original thrust-based fusion requires such an array).

Reply:
You're right, that implementation is really strange. `overlapper_triggered.cu` will hopefully be removed once the new overlapper is done.
I'm just thinking if this should somehow be more directly connected to `Overlap` object...